bricks.js

```diff
@@ -189,23 +192,64 @@ function drawpuzzle(nColumns, nRows) {
 	drawgaps(nColumns, nRows)
 }
 
-drawpuzzle(nColumns, nRows);
-
-const puzzle1 = {'B1-1': 3, 'B4-1': 1, 'B2-2': 3, 'B1-3': 3, 'B4-3': 4};
-
 function drawpuzzletext(puzzle) {
 	for (let key in puzzle) {
 		const xcoord = parseInt(document.getElementById(key).getAttributeNS(null, 'x')) + brickwidth / 2;
 		const ycoord = parseInt(document.getElementById(key).getAttributeNS(null, 'y')) + .75 * brickheight;
 		const clue = document.createElementNS(svgns, 'text');
 		clue.setAttributeNS(null, 'x', xcoord);
 		clue.setAttributeNS(null, 'y', ycoord);
+		clue.setAttributeNS(null, 'class', 'clue');
 		clue.setAttributeNS(null, 'font-size', .75 * brickheight);
-		clue.setAttributeNS(null, 'text-anchor', 'middle');
-		clue.setAttributeNS(null, 'font-family', 'Helvetica');
 		clue.innerHTML = puzzle[key];
-		document.getElementById('canvas').appendChild(clue);
+		document.getElementById('canvas').appendChild(clue); 
+	}
+}
+/* determines if the currently 'clicked' elements form one continuous loop with no stray elements */
+function isLoop() {
+	let clickedelems = Array.from(document.getElementsByClassName('clicked'));
```

Comment:
a style thing: use so-called "camelCase" with identifiers like "clickedelems" that are multiple words. So "clickedElems".